### ISSUE TITLE
Checks running in parallel and causing issues

### ISSUE THREAD

**Issue by grappler** [WPTRT/theme-sniffer#77]
At the moment when a check is running on a theme you can start a second check in parallel by selecting another theme from the dropdown and clicking go.

This causes issues like gettings results for two themes and the progress bar goes over 100%.

I see two solutions here.
- Remove the go button when running checks and add it back once the checks has reached 100%.
- Stop the current theme checks when another theme check is started.

@dingo-d Would love your insight as you wrote most of the JS code.

**Comment by carolinan** [comment]
Same as issue 71

**Comment by dingo-d** [comment]
If you want to I can disable the button on check start, in the #74 PR, and then enable it on the finish. Or I can open a new PR once that has been merged.

**Comment by grappler** [comment]
@dingo-d A separate PR would be better. We could discuss if a stop button is possible too.

**Comment by dingo-d** [comment]
Sure thing, a stop button shouldn't be hard to find. And all that is needed is to add a button, bind ajax call to a variable (request) and then use `request.abort();` on button click.

**Comment by dingo-d** [comment]
I've added a stop button that is working, but the sniff in progress will still output the last report of the file that was being sniffed on theme switch. I'll optimise this in the next release.

**Comment by dingo-d** [comment]
I'll close this as the new release will be a bit different and will implement new cool stuff from the phpcs v3.x 🙂
